rir/src/compiler/pir/instruction.cpp

```diff
@@ -62,7 +69,65 @@ void printPaddedTypeAndRef(std::ostream& out, const Instruction* i) {
     }
 }
 
-bool Instruction::validIn(Code* code) const { return bb()->owner == code; }
+void Instruction::printEffects(std::ostream& out, bool tty) const {
+    Effects eff = effects;
+    if (tty) {
+        if (eff.contains(Effect::WritesEnv))
```

Comment:
why is this here? It does not make sense to change the color of the effects. the idea is that the environment itself is printed in color, when it is changed.